request.go

```diff
@@ -1,11 +1,40 @@
 package sarama
 
+import (
+	"fmt"
+)
+
 type requestEncoder interface {
 	encoder
 	key() int16
 	version() int16
 }
 
+type requestHeader struct {
```

Comment:
I think I'd rather touch production code to get the better structure, our test suite will definitely catch any regressions that might be introduced.